### PR TITLE
Fix signal propagation from dockerfy to its start and primary commands.

### PR DESCRIPTION
**Pull request by markriggins**
@lee can you take a look at this?

### CODE REVIEW

File: exec.go [hunk 1]
```diff
@@ -43,27 +43,42 @@ func runCmd(ctx context.Context, cancel context.CancelFunc, cmd *exec.Cmd, cance
...
-			if debugFlag {
-				log.Printf("Command `%s` done waiting for signals (ctx.Done())", toString(cmd))
-			}
-			signalProcessWithTimeout(cmd, syscall.SIGTERM)
-			// already cancelled
-		}
+        for {
+            select {
+            case sig, ok := <-sigs:
+                if !ok {
```
Comment (markriggins) [inline exec.go]: once the sigs channel closes, we should stop this goroutine

File: exec.go [hunk 2]
```diff
@@ -43,27 +43,42 @@ func runCmd(ctx context.Context, cancel context.CancelFunc, cmd *exec.Cmd, cance
...
+                signalProcessWithTimeout(cmd, syscall.SIGKILL)
+                return
+            }
+        }
 	}()
 
 	err = cmd.Wait()
-	close(sigs)
+    signal.Stop(sigs)
+    close(sigs)
```
Comment (markriggins) [inline exec.go]: signal.Stop(sigs) tells signal to stop sending signals on this channel, which avoids a panic in certain edge cases where the signal arrives after we've closed

**Comment by leetreveil** [comment]
What's the return code when we send the `INT` signal to the dockerfy process?

File: exec.go [hunk 3]
```diff
@@ -74,15 +89,23 @@ func runCmd(ctx context.Context, cancel context.CancelFunc, cmd *exec.Cmd, cance
...
+		if exitCode == 0 {
+			// First child to exit with an error sets the exitCode
+			if exiterr, ok := err.(*exec.ExitError); ok {
+				if waitStatus, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+					exitCode = waitStatus.ExitStatus()
+					if verboseFlag {
+						log.Printf("\tand exit_code %d", exitCode)
+					}
+				} else {
+					log.Printf("Cloud not determine WaitStatus")
```
Comment (leetreveil) [inline exec.go]: Could

Comment (markriggins) [inline exec.go]: Ty.  Cloud computing on the brain LOL

Comment (leetreveil) [inline exec.go]: I often type mysql instead of myself :/

**Pull request merged**